src/Theme.cpp

```diff
@@ -0,0 +1,411 @@
+#include "stdafx.h"
+
+#include <fmt/core.h>
+#include <scn/scn.h>
+
+#include "Utils.h"
+
+#include "Theme.h"
+
+using nlohmann::json;
+
+// Helpers
+
+static json XyDimensionsToObject(const ImVec2& aXy)
+{
+    return json::object({{"x", aXy.x}, {"y", aXy.y}});
+}
+
+static ImVec2 XyDimensionsFromObject(const json& aJson, const ImVec2& aDefaults)
+{
+    return ImVec2(aJson.value("x", aDefaults.x), aJson.value("y", aDefaults.y));
+}
+
+static json DirectionToLabel(const ImGuiDir& aDirection)
+{
+    switch (aDirection)
+    {
+    case ImGuiDir_None: return "None";
+    case ImGuiDir_Left: return "Left";
+    case ImGuiDir_Right: return "Right";
+    case ImGuiDir_Up: return "Up";
+    case ImGuiDir_Down: return "Down";
+    default: return "Unknown";
+    }
+}
+
+static ImGuiDir DirectionFromLabel(const std::string& aDirectionLabel, const ImGuiDir& aDefaultDirection)
+{
+    if (aDirectionLabel == "None")
+        return ImGuiDir_None;
+    if (aDirectionLabel == "Left")
+        return ImGuiDir_Left;
+    if (aDirectionLabel == "Right")
+        return ImGuiDir_Right;
+    if (aDirectionLabel == "Up")
+        return ImGuiDir_Up;
+    if (aDirectionLabel == "Down")
+        return ImGuiDir_Down;
+
+    return aDefaultDirection;
+}
+
+static std::string ColorToHex(const ImVec4& aColor)
+{
+    return fmt::format(
+        //    R     G     B     A
+        "#{:02x}{:02x}{:02x}{:02x}", static_cast<int>(aColor.x * 255), static_cast<int>(aColor.y * 255), static_cast<int>(aColor.z * 255), static_cast<int>(aColor.w * 255));
+}
+
+static ImVec4 ColorFromHex(const std::string& aHex, const ImVec4& aDefaultColor)
+{
+    if (aHex.empty())
+    {
+        return aDefaultColor;
+    }
+
+    int r, g, b = 0;
```

Comment:
prefer cstdint types, try to avoid int short long etc.
i see bunch of these, please change these to int32_t int16_t etc.